underlay/src/main/java/bio/terra/tanagra/query/bigquery/BQExecutor.java

```diff
@@ -89,37 +93,49 @@ public SqlQueryResult run(SqlQueryRequest queryRequest) {
 
   public String export(
       SqlQueryRequest queryRequest,
-      String fileNamePrefix,
       String exportProjectId,
+      List<String> exportDatasetIds,
       List<String> exportBucketNames) {
-    // Validate the filename.
-    // GCS file name must be in wildcard format.
-    // https://cloud.google.com/bigquery/docs/reference/standard-sql/other-statements#export_option_list:~:text=The%20uri%20option%20must%20be%20a%20single%2Dwildcard%20URI
-    String validatedFilename;
-    if (fileNamePrefix == null || !fileNamePrefix.contains("*")) {
-      validatedFilename = "tanagra_" + System.currentTimeMillis() + "_*.csv";
-      LOGGER.warn(
-          "No (or invalid) filename specified for GCS export ({}), using default: {}",
-          fileNamePrefix,
-          validatedFilename);
-    } else {
-      validatedFilename = fileNamePrefix;
-    }
+    LOGGER.info("Exporting BQ query: {}", queryRequest.getSql());
+
+    // Create a temporary view with the results of the query.
+    final String tempTableName = TEMPORARY_TABLE_BASE_NAME + '_' + Instant.now().toEpochMilli();
```

Comment:
Would it be easy to also include a somewhat unique element of the request in here as well? Or nanoseconds? It's obviously not very likely to have a collision but worth decreasing the odds further if it's easy enough.

Reply:
Good idea. Added nanosec and a random int.